### ISSUE TITLE
Some combinations of features fail to build

### ISSUE THREAD

**Issue by Jess3Jane** [rust-lang/regex#769]
#### What version of regex are you using?

`1.5.2`, or with `regex-syntax` `0.6.24` with rust version `rustc 1.53.0-nightly (42816d61e 2021-04-24`

#### Describe the bug at a high level.

With default features disabled some feature combinations fail to compile.

#### What are the steps to reproduce the behavior?

For your `Cargo.toml`:

```toml
[package]
name="test"
version = "0.1.0"
edition = "2018"

[dependencies]
regex-syntax = { version = "0.6.24", default-features = false, features = ["unicode-perl"] }
```

This will also occur if you select the same feature set through `regex`:

```toml
[package]
name="test"
version = "0.1.0"
edition = "2018"

[dependencies]
regex = { version = "1.5.2", default-features = false, features = ["std", "unicode-perl"] }
```

#### What is the actual behavior?

```
  Compiling regex-syntax v0.6.24
error[E0433]: failed to resolve: use of undeclared crate or module `unicode_tables`
   --> /home/jess/.cargo/registry/src/github.com-1ecc6299db9ec823/regex-syntax-0.6.24/src/unicode.rs:351:13
    |
351 |         use unicode_tables::perl_space::WHITE_SPACE;
    |             ^^^^^^^^^^^^^^ use of undeclared crate or module `unicode_tables`

error[E0433]: failed to resolve: use of undeclared crate or module `unicode_tables`
   --> /home/jess/.cargo/registry/src/github.com-1ecc6299db9ec823/regex-syntax-0.6.24/src/unicode.rs:375:13
    |
375 |         use unicode_tables::perl_decimal::DECIMAL_NUMBER;
    |             ^^^^^^^^^^^^^^ use of undeclared crate or module `unicode_tables`

error[E0425]: cannot find value `WHITE_SPACE` in this scope
   --> /home/jess/.cargo/registry/src/github.com-1ecc6299db9ec823/regex-syntax-0.6.24/src/unicode.rs:352:22
    |
352 |         Ok(hir_class(WHITE_SPACE))
    |                      ^^^^^^^^^^^ not found in this scope
    |
help: consider importing this constant
    |
1   | use crate::unicode_tables::perl_space::WHITE_SPACE;
    |

error[E0425]: cannot find value `DECIMAL_NUMBER` in this scope
   --> /home/jess/.cargo/registry/src/github.com-1ecc6299db9ec823/regex-syntax-0.6.24/src/unicode.rs:376:22
    |
376 |         Ok(hir_class(DECIMAL_NUMBER))
    |                      ^^^^^^^^^^^^^^ not found in this scope
    |
help: consider importing this constant
    |
1   | use crate::unicode_tables::perl_decimal::DECIMAL_NUMBER;
    |

error: aborting due to 4 previous errors

Some errors have detailed explanations: E0425, E0433.
For more information about an error, try `rustc --explain E0425`.
error: could not compile `regex-syntax`
```

#### What is the expected behavior?

The program should compile.

**Comment by BurntSushi** [comment]
Ug. Thanks for the report. In my migration to Rust 2018, it has become apparent to me that `cargo fix` doesn't actually apply to all of the code. On top of that, my understanding is that [CI is specifically testing the case you've reported here](https://github.com/rust-lang/regex/blob/2393c5555cc499990ccf910ea35be5a694ae0ba9/regex-syntax/test#L13), so on top of `cargo fix` not doing what I thought it would do, I also appear to have a false confidence in CI coverage. :-/

I'll get this fixed ASAP.

**Comment by BurntSushi** [comment]
Arrgh. Turns out, CI [did catch these errors](https://github.com/rust-lang/regex/runs/2482218726?check_suite_focus=true), but the `test` script doesn't have `set -x` enabled, so it doesn't actually failed the build.

**Comment by BurntSushi** [comment]
This is fixed in `regex 1.5.3` on crates.io.
